File: 2.1 readability/readability.c
#include <cs50.h>
#include <string.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

// Índice Coleman-Liau
// indice = 0,0588 * L - 0,296 * S - 15,8
// L = Número médio de letras por 100 palavras.
// S = Número médio de frases por 100 palavras.

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Itera pelo texto, para contar
    // as variáveis usadas na fórmula.
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
        if (isspace(text[i]))
        {
            words += 1;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }

    words += 1;

    // Cálculo para obter o índice.
    float average_letters = (float) letters / words * 100;
    float average_sentences = (float) sentences / words * 100;
    float formula = 0.0588 * average_letters - 0.296 * average_sentences - 15.8;
    int index = round(formula);

    // Seleção e print da grade.
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: 6.5-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = 0
    words = 0
    sentences = 0

    for character in text:
        if character.isalpha():
            letters += 1

        if character.isspace():
            words += 1

        if character == '.' or character == '!' or character == '?':
            sentences += 1

    words += 1
    average_letters = letters / words * 100
    average_sentences = sentences / words * 100
    formula = 0.0588 * average_letters - 0.296 * average_sentences - 15.8

    grade = round(formula)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

main()